### ISSUE TITLE
Platformio CI throws a lot of "stray in program" on linking time, just in Travis, not locally

### ISSUE THREAD

**Issue by AlexFBP** [AsharaStudios/GuruxDLMS.c#5]
[GuruxDLMS.c](https://github.com/Gurux/GuruxDLMS.c) is a C library to access DLMS/COSEM compliant meters. They bring some examples for Arduino in the Arduino_IDE folder.

In our fork we initially created a project per folder (master branch) and in each folder, the platformio.ini file has as `lib_deps` the URL of the same repo, and this worked. But this focus induces to PlatformIO to `git clone` every time, so, trying alternatives to use the same folder...:

1. well by specifying the parent library folder in `lib_extra_dirs` ( [platformio.ini](https://github.com/AsharaStudios/GuruxDLMS.c/blob/e66e9186577c52a768c1ee497e1b6fe1a286f3b4/Arduino_IDE/client/platformio.ini) - [job 38.2](https://travis-ci.com/AsharaStudios/GuruxDLMS.c/jobs/167645562) ) - as `lib_extra_dirs` should be a path to a folder with libraries, why not to use the folder containing the cloned repo...
2. well by creating sym-links in the lib folder of both PlatformIO projects ( [commit 94091cd](https://github.com/AsharaStudios/GuruxDLMS.c/commit/94091cd436971b0c400fa1ee2aac58b80f308f73) - [build 9](https://travis-ci.com/AsharaStudios/GuruxDLMS.c/builds/95945558) )
3. well by specifying `--lib="."` when running `platformio ci`, as suggested in the second template inside the generated .travis.yml ( [branch platformioUnify](https://github.com/AsharaStudios/GuruxDLMS.c/commits/platformioUnify) - [job 59.2](https://travis-ci.com/AsharaStudios/GuruxDLMS.c/jobs/170054751) ) because as it is, the project must be considered as a lbrary.

...gives you in Travis CI a LOT of errors at linking time ( [log for case 1](https://travis-ci.com/AsharaStudios/GuruxDLMS.c/jobs/167645562#L1517) - [log for case 2](https://travis-ci.com/AsharaStudios/GuruxDLMS.c/jobs/167519706#L643) - [log for case 3](https://travis-ci.com/AsharaStudios/GuruxDLMS.c/jobs/170054751#L1515) ) BUT, not if you clone the repo and do the same steps as in .travis.yml, PlatformIO section.

```
...
.pioenvs/mega/libd43/libGuruxDLMS.c:2:1: error: stray '`' in program
/               1547397696  0     0     0       13998     `
^
...
^
.pioenvs/mega/libd43/libGuruxDLMS.c:3:3: error: stray '\2' in program
...
var_toDouble var_print va_print //                                              40        `
^
.pioenvs/mega/libd43/libGuruxDLMS.c:3:3: error: stray '\202' in program
.pioenvs/mega/libd43/libGuruxDLMS.c:3:5: warning: null character(s) ignored
�  7V  7V  7V  7V  7V  7V  7V  7V  7V  7V  7V  7V  7V  7V  7V 
```

~~When you clone the repo locally those errors don't appear (Or at least by me, I'm using elementary OS 0.4, Ubuntu 16.04 based, zsh 5.1.1, platformio v3.6, pip v18.1, vscode v1.30.2)~~ (don't know why it keeps compiling in the original folder I was working, cloned in other folder and failed)

Does anybody know how to remove those warnings? The only way that the projects compile in Travis (at time and with my tested .travis.yml) is none of the mentioned in the list, just in the introduction of this issue: by specifying the URL of the same repo in platformio.ini `lib_deps`

**Comment by AlexFBP** [comment]
I saw that if i clone in other folder, those errors start to appear.

Logs for 4a9fc29 as compiled in my PC current working dir:

- [stdoutCompila.log](https://github.com/AsharaStudios/GuruxDLMS.c/files/2754023/stdoutCompila.log)
- [stderrCompila.log](https://github.com/AsharaStudios/GuruxDLMS.c/files/2754024/stderrCompila.log)
- [stdoutAndstderrCompila.log](https://github.com/AsharaStudios/GuruxDLMS.c/files/2754032/stdoutAndstderrCompila.log)

Same repo, same commit, but cloned again in other folder:

- [stdoutFalla.log](https://github.com/AsharaStudios/GuruxDLMS.c/files/2754035/stdoutFalla.log)
- [stderrFalla.log.zip](https://github.com/AsharaStudios/GuruxDLMS.c/files/2754038/stderrFalla.log.zip) - 57.4MB of (compressed) log? wtf?
- [stdoutAndstderrFalla.log.zip](https://github.com/AsharaStudios/GuruxDLMS.c/files/2754040/stdoutAndstderrFalla.log.zip) - 42.6 MB of (compressed) log...

**Comment by sytabaresa** [comment]
Hola :heart:

**Comment by sytabaresa** [comment]
Hola :heart:

**Comment by AlexFBP** [comment]
Fixed in d08982b5, `.travis.yml` to use Gurux in PlatformIO CI, rewriten to:

```yml
install:
...
- platformio lib -g install file://$(pwd)
script:
- platformio ci -C Arduino_IDE/client/platformio.ini Arduino_IDE/client
- platformio ci -C Arduino_IDE/server/platformio.ini Arduino_IDE/server
```
